schism/compiler.ss

```diff
@@ -695,6 +705,18 @@
   ;; intrinsics.  This pass simplifies away the introduced complexity
   ;; where it's not needed.
 
+  (define (effect-free? expr)
+    (or (literal? expr) (eq? (car expr) 'lambda)))
+  (define (simplify-seq head tail)
+    (cond
+     ((effect-free? head) tail)
+     ((eq? (car tail) 'seq)
+      ;; Quadratic, but shouldn't be hit normally.
+      (simplify-seq (simplify-seq head (cadr tail)) (caddr tail)))
```

Comment:
This only gets hit if the frontend puts a seq in a tail, right? Should we just make this case an error instead?

Reply:
Generally yes.  However in the presence of beta reductions / inlining I think we don't preserve this invariant:

```
(begin A ((lambda () (begin C D))))
=> (seq A (call (lambda () (seq C D))))
=> (seq A (seq C D))
```

So, to me it seems reasonable that we can be lenient here.

Reply:
Sounds good to me.